### PR TITLE
DOC: Adopt a harmonic color scheme with the dark mode of pydata-sphinx-theme

### PR DESCRIPTION
**Pull request by noritada**
This PR provides a simple solution to the 1st and 2nd issues documented in #22408.

Specifically, the body text color setting has been removed from numpy.css so that the setting in pydata-sphinx-theme is used. Also, the label color of "dev" version selection button is set to `#4a4a4a`, which was originally used as the body text color.

Since the 3rd issue may be a matter of taste and other section heading colors might need to be changed as well, it is left as is here at this moment.

The choice of colors may depend on the project policy and personal preferences, and I am not a professional colorist. So, any suggestions would be greatly appreciated.
Many thanks.

### CODE REVIEW

**Comment by melissawm** [comment]
Thanks, @noritada - I'll ping @tupui to take a look. I'm pretty sure if we do this, we should also do it for SciPy, right?

**Comment by mattip** [comment]
Would this be better suited to the upstream repo? I seem to recall we had patched the theme colors because upstream's first releases were not what we wanted, but maybe by now they have fixed some of these problems. I also see pydata-sphinx-theme [released 0.11](https://github.com/pydata/pydata-sphinx-theme/releases) two days ago, maybe worth looking at updating our version of the theme and then revisiting this css file for possible deletions.

**Comment by tupui** [comment]
Thanks for the ping @melissawm. I will have a look on Monday. Yes we want to be in sync with NumPy.

@mattip for the theme, yes they released but still did not fix the template issue. I did not check here, but if you have custom templates for e.g. classes then build time doubles. This appeared in 0.10 and they still haven't figured out a fix. I just could narrow it down to know it's due to the templates.

**Comment by noritada** [comment]
Thanks everyone for providing comments and sharing background information. I will look forward to future reviews and further discussion.

File: doc/source/_static/numpy.css [hunk 1]
```diff
@@ -48,6 +47,7 @@ Nat Methods 8, 441 (2011). https://doi.org/10.1038/nmeth.1618
 #version_switcher_button[data-active-version-name*="dev"] {
   background-color: #E69F00;
   border-color: #E69F00;
+  color:#4A4A4A; /* numpy.org body color */
```
Comment (InessaPawson) [inline doc/source/_static/numpy.css:50]:
```suggestion
  color:#6C7A89; /* NumPy Slate Gray */
```

@noritada, thank you for sharing your thoughts! These are all valid concerns. For the version selection button in the dark mode, I suggest using NumPy Slate Gray (#6C7A89) from the NumPy color palette (https://github.com/numpy/numpy.org#numpy-color-palette).

Comment (noritada) [reply doc/source/_static/numpy.css]: @InessaPawson Thank you for the suggestion! I must have missed that color palette.
I have pushed the suggested color update.

**Comment by tupui** [comment]
I had a look. The contrast is indeed better, but pure black might just be better here no (as in non-dark mode)?

Screenshot 2022-10-10 at 16 50 19
Screenshot 2022-10-10 at 16 49 28


The new version of the theme does have better defaults now, but since here we have custom colours for the dev/stable/PR version, it needs to be manually adjusted. And since it's project specific, nothing to upstream there.

**Comment by noritada** [comment]
@tupui Yes, black or a similar color would look better in terms of contrast with the orange. I also agree that the color of the version selection button is project specific and does not need to be discussed with upstream.

**Comment by seberg** [comment]
@noritada do you want to just make it black here, then?  It sounds like this PR should be otherwise.  Or if you prefer merge as is, I am happy either way.

**Comment by noritada** [comment]
@seberg Thank you very much. I was not sure whose opinion I should respect. I think black is better, so I have pushed the update.

**Comment by seberg** [comment]
Thanks everyone.  Yeah, if anyone has a stronger opinion about it, we can still adapt the colorscheme.

**Pull request merged**